sdk/python/kfp/compiler/compiler_utils_test.py

```diff
@@ -106,7 +106,68 @@ def test_additional_input_name_for_pipeline_channel(self, channel,
                 },
                 'replica_count': 1
             }],
-        },)
+        },
+        {
+            'data': [{
+                'first_name':
+                    f'my_first_name: {str(pipeline_channel.PipelineParameterChannel(name="Output", channel_type="String", task_name="first_name"))}',
+                'last_name':
+                    f'my_last_name: {str(pipeline_channel.PipelineParameterChannel(name="Output", channel_type="String", task_name="last_name"))}',
+            }],
+            'old_value':
+                '{{channel:task=first_name;name=Output;type=String;}}',
+            'new_value':
+                '{{$.inputs.parameters['
+                'pipelinechannel--first_name-Output'
+                ']}}',
+            'expected': [{
+                'first_name':
+                    'my_first_name: {{$.inputs.parameters[pipelinechannel--first_name-Output]}}',
+                'last_name':
+                    f'my_last_name: {str(pipeline_channel.PipelineParameterChannel(name="Output", channel_type="String", task_name="last_name"))}',
+            }],
+        },
+        {
+            'data': [{
+                'project': 'project',
+                'location': 'US',
+                'job_configuration_query': {
+                    'query': 'SELECT * FROM `project.dataset.input_table`',
+                    'destinationTable': {
+                        'projectId':
+                            'project',
+                        'datasetId':
+                            'dataset',
+                        'tableId':
+                            f'output_table_{str(pipeline_channel.PipelineParameterChannel(name="Output", channel_type="String", task_name="table_suffix"))}',
+                    },
+                    'writeDisposition': 'WRITE_TRUNCATE',
+                },
+            }],
+            'old_value':
+                '{{channel:task=table_suffix;name=Output;type=String;}}',
+            'new_value':
+                '{{$.inputs.parameters['
+                'pipelinechannel--table_suffix-Output'
+                ']}}',
+            'expected': [{
+                'project': 'project',
+                'location': 'US',
+                'job_configuration_query': {
+                    'query': 'SELECT * FROM `project.dataset.input_table`',
+                    'destinationTable': {
+                        'projectId':
+                            'project',
+                        'datasetId':
+                            'dataset',
+                        'tableId':
+                            'output_table_{{$.inputs.parameters[pipelinechannel--table_suffix-Output]}}',
+                    },
+                    'writeDisposition': 'WRITE_TRUNCATE',
+                },
+            }],
+        },
+    )
```

Comment:
This test case partially replaces the value in the second level. It fails by the original implementation.